src/org/openstreetmap/josm/gui/tagging/presets/PresetSearchFilter.java

```diff
@@ -0,0 +1,72 @@
+// License: GPL. For details, see LICENSE file.
+package org.openstreetmap.josm.gui.tagging.presets;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+import static org.openstreetmap.josm.tools.I18n.marktr;
+
+/**
+ * This enum defines different filters for searching presets.
+ */
+public enum PresetSearchFilter {
+    ONLY_APPLICABLE(marktr("Show only applicable to selection")),
+    SEARCH_IN_TAGS(marktr("Search in tags")),
+    DEPRECATED_TAGS(marktr("Show deprecated tags"));
+
+    /**
+     * Map containing the preferences for the filters
+     */
+    private Map<PresetSearchFilter, Boolean> filtersPreference;
+
+    static {
+        for (PresetSearchFilter filter : values()) {
+            filter.filtersPreference = new EnumMap<>(PresetSearchFilter.class);
+            filter.filtersPreference.put(filter, true);
+        }
+    }
+
+    /**
+     * Sets the preference for the filter
+     * @param filter the filter to set the preference for
+     * @param pref true if the filter is enabled, false otherwise
+     * @since xxx
+     */
+    public void setPref(PresetSearchFilter filter, Boolean pref) {
+        filtersPreference.put(filter, pref);
+    }
+
+    /**
+     * Gets the preference for the filter
+     * @param filter the filter to get the preference for
+     * @return true if the filter is enabled, false otherwise
+     * @since xxx
+     */
+    public Boolean getPref(PresetSearchFilter filter) {
+        return filtersPreference.get(filter);
```

Comment:
This should be done via the standard `Config` interface _or_ one of the `Property` classes.